tests/l2/test_l2_configure.py

```diff
@@ -0,0 +1,161 @@
+]
+
+
+@pytest.fixture(autouse=True)
+def setup_env(duthosts, rand_one_dut_hostname):
+    """
+    Setup/teardown fixture for each loopback interface test.
+    rollback to check if it goes back to starting config
+
+    Args:
+        duthosts: list of DUTs.
+        rand_selected_dut: The fixture returns a randomly selected DuT.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+    duthost.shell("sudo cp {} {}".format(CONFIG_DB, CONFIG_DB_BAK))
+
+    yield
+
+    duthost.shell("sudo cp {} {}".format(CONFIG_DB_BAK, CONFIG_DB))
+    duthost.shell("sudo rm -f {}".format(CONFIG_DB_BAK))
+    config_reload(duthost)
+    wait_critical_processes(duthost)
+
+
+def is_table_empty(duthost, table):
+    """
+    @summary: Verify a table is empty.
+
+    Args:
+        duthost: DUT host object.
+        table: Table name to verify.
+    """
+    # grep returns 1 when there is no match, use || true to override that.
+    count = int(
+        duthost.shell(
+            'sonic-db-cli CONFIG_DB KEYS "{}|*" | grep -c {} || true'.format(
+                table, table
+            )
+        )["stdout"]
+    )
+    return count == 0
+
+
+def get_db_version(duthost):
+    """
+    @summary: Get the current database version.
+
+    Args:
+        duthost: DUT host object.
+    """
+    try:
+        return duthost.shell('sonic-db-cli CONFIG_DB HGET "VERSIONS|DATABASE" VERSION')[
+            "stdout"
+        ]
+    except Exception as e:
+        logger.error("Failed to get database version: {}".format(e))
+        return ""
+
+
+def test_no_hardcoded_minigraph(duthosts, rand_one_dut_hostname, tbinfo):
+    """
+    @summary: A testcase asserts no hardcoded minigraph config is imported to config_db during L2 configuration.
+
+    Args:
+        duthosts: list of DUTs.
+        rand_one_dut_hostname: The fixture returns a randomly selected DuT.
+        tbinfo: The testbed information. Needed for configuring management interface.
+
+    """
+    # Setup.
+    duthost = duthosts[rand_one_dut_hostname]
+    if is_table_empty(duthost, "TELEMETRY") or is_table_empty(duthost, "RESTAPI"):
+        pytest.skip("TELEMETRY or RESTAPI table is empty. Please load minigraph first.")
+
+    hwsku = duthost.facts["hwsku"]
+    mgmt_fact = duthost.get_extended_minigraph_facts(tbinfo)["minigraph_mgmt_interface"]
+
+    # Step 2: Configure DUT into L2 mode.
+    # Save original config
+    duthost.shell("sudo cp {} {}".format(CONFIG_DB, CONFIG_DB_BAK))
+    # Perform L2 configuration
+    L2_INIT_CFG_FILE = "/tmp/init_l2_cfg.json"
+    MGMT_CFG_FILE = "/tmp/mgmt_cfg.json"
+    L2_CFG_FILE = "/tmp/l2_cfg.json"
+    gen_l2_cfg = "sudo sonic-cfggen --preset l2 -p -H -k {} > {}".format(
+        hwsku, L2_INIT_CFG_FILE
+    )
+    duthost.shell(gen_l2_cfg)
+    gen_mgmt_cfg = """
+    echo '
+    {{
+        "MGMT_INTERFACE": {{
+            "eth0|{}/{}": {{
+                "gwaddr": "{}"
+            }}
+        }},
+        "DEVICE_METADATA": {{
```

Comment:
>DEVICE_METADATA [](http://example.com/codeflow?start=9&length=15)

The step in test does not follow [https://github.com/sonic-net/SONiC/wiki/L2-Switch-mode](https://github.com/sonic-net/SONiC/wiki/L2-Switch-mode) exactly. For example, DualToR is not considered. And the commands used does not exact match.

Reply:
For the record, this comes from [testbed_set_l2.yml](https://github.com/sonic-net/sonic-mgmt/blob/master/ansible/testbed_set_l2_mode.yml). The wiki way requires running over console and I haven't find a reliable API in sonic-mgmt that pass the test. I'll see if there is any other way to make it work.

Reply:
Spoke with Gang and Xin yesterday. And it appears the [testbed_set_l2.yml](https://github.com/sonic-net/sonic-mgmt/blob/master/ansible/testbed_set_l2_mode.yml) method of setting up is a L2 mode is a workaround since ansible library requires the ssh session to be always active, and we don't have library support for doing it through console.